app/controllers/api/v1/statistics_controller.rb

```diff
@@ -16,24 +16,46 @@ def index
       private
 
       def number_of_students
-        # The same as:
-        # Role.where(group: Person::STUDENT).map{|x| x.people ? x.people : nil}.flatten.compact.count,
-        Role.joins(:people).where(group: Person::STUDENT).count
+        number_of_informed_consents_given(Rails.application.config.settings.protocol_names&.student)
       end
 
       def number_of_mentors
-        # The same as:
-        # Role.where(group: Person::MENTOR).map{|x| x.people ? x.people : nil}.flatten.compact.count
-        Role.joins(:people).where(group: Person::MENTOR).count
+        number_of_informed_consents_given(Rails.application.config.settings.protocol_names&.mentor)
+      end
+
+      def number_of_informed_consents_given(protocol_names)
+        # If this method is used to count people then it is assumed that a person is asked
+        # for informed consent in at most one of her protocols.
+        return 0 if protocol_names.blank? # In case the `protocol_names` setting was nil
+
+        protocol_ids = protocol_names.map do |protocol_name|
+          Protocol.find_by_name(protocol_name)&.id
+        end.compact
+        return 0 if protocol_ids.blank?
+
+        ProtocolSubscription.where('informed_consent_given_at IS NOT NULL').where(protocol_id: protocol_ids).count
```

Comment:
Dit gaat niet helemaal goed als we een project hebben waarbij docenten en studenten een zelfde protocol volgen.

Reply:
Fixed. Ik heb het helemaal herschreven.